sycl/source/queue.cpp

```diff
@@ -312,6 +312,57 @@ event queue::submit_with_event_impl(
   return impl->submit_with_event(CGH, SubmitInfo, CodeLoc, IsTopCodeLoc);
 }
 
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+event queue::submit_direct_with_event_impl(
```

Comment:
Having overloads vs templates is important for public APIs (because of the implicit conversion), but for our implementation details we can just use template to reduce amount of boilerplate code: https://godbolt.org/z/rPW4jx8h7

Headers will only have template declaration, .cpp file will export necessary instantiations.